src/java.base/share/classes/java/util/random/RandomGenerator.java

```diff
@@ -0,0 +1,1546 @@
+     * However, each generator produced also has type
+     * {@link ArbitrarilyJumpableGenerator}; with care, different jump distances
+     * can be used to traverse the entire state cycle in various ways.
+     *
+     * <p> Objects that implement {@link ArbitrarilyJumpableGenerator} are
+     * typically not cryptographically secure. Consider instead using
+     * {@link SecureRandom} to get a cryptographically secure pseudo-random
+     * number generator for use by security-sensitive applications.
+     */
+    interface ArbitrarilyJumpableGenerator extends LeapableGenerator {
+
+        /**
+         * Returns an instance of {@link ArbitrarilyJumpableGenerator} that
+         * utilizes the {@code name} <a href="package-summary.html#algorithms">algorithm</a>.
+         *
+         * @param name  Name of random number generator
+         *              <a href="package-summary.html#algorithms">algorithm</a>
+         *
+         * @return An instance of {@link ArbitrarilyJumpableGenerator}
+         *
+         * @throws NullPointerException if name is null
+         * @throws IllegalArgumentException if the named algorithm is not found
+         */
+        static ArbitrarilyJumpableGenerator of(String name) {
+            Objects.requireNonNull(name);
+
+            return RandomGeneratorFactory.of(name, ArbitrarilyJumpableGenerator.class);
+        }
+
+        /**
+         * Returns a new generator whose internal state is an exact copy of this
+         * generator (therefore their future behavior should be identical if
+         * subjected to the same series of operations).
+         *
+         * @return a new object that is a copy of this generator
+         */
+        ArbitrarilyJumpableGenerator copy();
+
+        /**
+         * Alter the state of this pseudorandom number generator so as to jump
+         * forward a distance equal to 2<sup>{@code logDistance}</sup> within
+         * its state cycle.
+         *
+         * @param logDistance the base-2 logarithm of the distance to jump forward within the state
+         *                    cycle
+         *
+         * @throws IllegalArgumentException if {@code logDistance} is
+         *                                  2<sup>{@code logDistance}</sup> is
+         *                                  greater than the period of this generator
+         */
+        void jumpPowerOfTwo(int logDistance);
+
+        /**
+         * Alter the state of this pseudorandom number generator so as to jump
+         * forward a specified distance within its state cycle.
+         *
+         * @param distance the distance to jump forward within the state cycle
+         *
+         * @throws IllegalArgumentException if {@code distance} is not greater than
+         *                                  or equal to 0.0, or is greater than the
+         *                                  period of this generator
+         */
+        void jump(double distance);
```

Comment:
Suggest Override and inheritDoc combo.

Reply:
jump(double distance) does not occur in superinterface. The ability to jump arbitrarily is specific to ArbitrarilyJumpableGenerator.